src/promptflow-core/promptflow/core/_connection_provider/_workspace_connection_provider.py

```diff
@@ -333,8 +339,82 @@ def _convert_to_connection_dict(cls, conn_name, conn_data):
                 name=conn_name,
             )
 
-    def list(self):
-        raise NotImplementedError("Method 'list' is not implemented now.")
+    @classmethod
+    def _build_list_connection_dict(
+        cls, subscription_id, resource_group_name, workspace_name, credential
+    ) -> Dict[str, dict]:
+        url = LIST_CONNECTION_URL.format(
+            sub=subscription_id,
+            rg=resource_group_name,
+            ws=workspace_name,
+        )
+        # Note: There is a try-catch in get arm token. It requires azure-ai-ml.
+        # TODO: Remove the azure-ai-ml dependency.
+        from ._utils import get_arm_token
+
+        try:
+            from azure.core.exceptions import ClientAuthenticationError
+
+            from ._models import WorkspaceConnectionPropertiesV2BasicResourceArmPaginatedResult
+        except ImportError as e:
+            raise MissingRequiredPackage(
+                message="Please install 'promptflow-core[azureml-serving]' to use workspace connection."
```

Comment:
Is this the same message we present when user do pf config set connection.provider=azureml://xx?